private/buf/bufcli/bufcli.go

```diff
@@ -549,8 +549,7 @@ func NewConfig(container appflag.Container) (*bufapp.Config, error) {
 	return bufapp.NewConfig(container, externalConfig)
 }
 
-// NewRegistryProvider creates a new registryv1alpha1apiclient.Provider.
-func NewRegistryProvider(ctx context.Context, container appflag.Container) (registryv1alpha1apiclient.Provider, error) {
+func newGRPCRegistryProvider(ctx context.Context, container appflag.Container) (registryv1alpha1apiclient.Provider, error) {
```

Comment:
I removed it in this PR bc it was unused and wouldn't pass lint.  I didn't want to make it exported to pass so I will paste here instead:

When the Connect switch is officially turned on in bufcli, this function will be replaced by `newConnectRegistryProvider` like so:

```
func newConnectRegistryProvider(ctx context.Context, container appflag.Container) (registryv1alpha1apiclient.Provider, error) {
	client := http2client.NewClient(
		http2client.WithObservability(),
	)
	options := []bufapiclient.RegistryProviderOption{
		bufapiclient.RegistryProviderWithContextModifierProvider(NewContextModifierProvider(container)),
		bufapiclient.RegistryProviderWithAddressMapper(func(address string) string {
			if buftransport.IsAPISubdomainEnabled(container) {
				address = buftransport.PrependAPISubdomain(address)
			}
			return buftransport.PrependHTTPS(address)
		}),
	}
	return bufapiclient.NewConnectClientProvider(container.Logger(), client, options...)
}
```